lib/ansible_tower_client/base_models/job_template.rb

```diff
@@ -25,5 +37,20 @@ def extra_vars_hash
     def override_raw_attributes
       { :credential => :credential_id, :inventory => :inventory_id, :project => :project_id }
     end
+
+    private
+
+    def check_required_attributes_for_options(options)
```

Comment:
Maybe something like this would be better?
```ruby
def validate_launch_options(options)
  errors = []
  REQUIRED_ATTRIBUTES_FOR_OPTIONS.each do |option, checkmark|
    next unless (options.key?(option.to_sym) || options.key?(option.to_s)
    next if (respond_to?(checkmark) && send(checkmark))
    errors << "#{option} included but #{checkmark} was false"
  end

  return if errors.empty?

  raise ArgumentError, "Unused option(s): #{errors.join(", ")}"
end

```

Reply:
It also removes the need to Object#try, Object#blank and Hash#slice

Reply:
Some AS methods was already required via something else, so I'll just remove `try` and `to_sentence`.